src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -305,14 +339,14 @@ private int RunCore(TextWriter consoleOutput, CancellationToken cancellationToke
                 return Succeeded;
             }
 
-            if (PrintErrors(Arguments.Errors, consoleOutput))
+            if (ReportErrors(Arguments.Errors, consoleOutput, errorLogger))
```

Comment:
consoleoutput and errorlogger seems both are just logger that doesnt have same abstraction above it.